### ISSUE TITLE
`constant_identifier_names` false positive for destructured field names

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4297]
Follow-up from #4295.

We should see no lints here.

```dart
class A {
  var a_b;
}

f(A a) {
  switch (a) {
    case A(:var a_b):
  }
  switch (a) {
    case A(:var a_b?):
    case A(:var a_b!): 
  }
}
```
